--- source/libvpx.orig/vp8/encoder/onyx_if.c	2010-09-10 10:33:18.000000000 -0400
+++ source/libvpx/vp8/encoder/onyx_if.c	2010-09-23 20:43:27.000000000 -0400
@@ -2678,6 +2678,7 @@ int vp8_update_entropy(VP8_PTR comp, int
 
 void vp8_write_yuv_frame(const char *name, YV12_BUFFER_CONFIG *s)
 {
+#if 0
     FILE *yuv_file = fopen(name, "ab");
     unsigned char *src = s->y_buffer;
     int h = s->y_height;
@@ -2710,6 +2711,7 @@ void vp8_write_yuv_frame(const char *nam
     while (--h);
 
     fclose(yuv_file);
+#endif
 }
 
 static void scale_and_extend_source(YV12_BUFFER_CONFIG *sd, VP8_COMP *cpi)
@@ -4354,13 +4356,14 @@ static void encode_frame_to_data_rate(VP
 
     if (cm->frame_type == KEY_FRAME)
         cm->refresh_last_frame = 1;
-
+#if 0
     if (0)
     {
         FILE *f = fopen("gfactive.stt", "a");
         fprintf(f, "%8d %8d %8d %8d %8d\n", cm->current_video_frame, (100 * cpi->gf_active_count) / (cpi->common.mb_rows * cpi->common.mb_cols), cpi->this_iiratio, cpi->next_iiratio, cm->refresh_golden_frame);
         fclose(f);
     }
+#endif
 
     // For inter frames the current default behaviour is that when cm->refresh_golden_frame is set we copy the old GF over to the ARF buffer
     // This is purely an encoder descision at present.
@@ -4730,7 +4733,7 @@ static void encode_frame_to_data_rate(VP
     // reset to normal state now that we are done.
 
 
-
+#if 0
     if (0)
     {
         char filename[512];
@@ -4744,7 +4747,7 @@ static void encode_frame_to_data_rate(VP
 
     // DEBUG
     //vp8_write_yuv_frame("encoder_recon.yuv", cm->frame_to_show);
-
+#endif
 
 }
 
