diff --git a/build/make/obj_int_extract.c b/build/make/obj_int_extract.c
index c46d9d5..bb3588b 100644
--- a/build/make/obj_int_extract.c
+++ b/build/make/obj_int_extract.c
@@ -535,7 +535,8 @@ int parse_elf(uint8_t *buf, size_t sz, output_fmt_t mode)
 
                 strcpy(strtsb_name, (char *)(elf.buf + shdr.sh_offset + shdr.sh_name));
 
-                if (!(strcmp(strtsb_name, ".shstrtab")))
+                if (!(strcmp(strtsb_name, ".shstrtab")) 
+                    || (strcmp(strtsb_name, ".strtab")))
                 {
                     /* log_msg("found section: %s\n", strtsb_name); */
                     strtab_off64 = shdr.sh_offset;
